### PR TITLE
Add docker workflow for Apple M1 architecture

### LINKED ISSUE #273: Update release workflow to support more platform
`sbt docker` does not build images for arches like Apple M1. Let's use the same workflow as `alephium`.

ref: https://github.com/alephium/alephium/blob/master/.github/workflows/docker-buildx.yml

### PR DESCRIPTION
**Pull request by tdroxler**
Resolves  #273

### CODE REVIEW

File: .github/workflows/docker-buildx.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        if: ${{ !contains(steps.get_version.outputs.VERSION, '-') }}
+        uses: docker/build-push-action@v2
+        with:
+          context: docker/release
+          file: ./docker/release/Dockerfile.release
+          build-args: RELEASE=${{ steps.get_version.outputs.VERSION-NO-V }}
+          platforms: linux/amd64,linux/arm64,linux/arm/v7
+          tags: |
+            docker.io/alephium/explorer-backend:${{ steps.get_version.outputs.VERSION }}
+            docker.io/alephium/explorer-backend:latest
```
Comment (polarker) [inline .github/workflows/docker-buildx.yml:49]: There will be a race condition if this pushes at the same time with `docker.yml`.

Let's change this line in `build.sbt`:
```
-      val baseImageName = "alephium/explorer-backend"
+     val baseImageName = "alephium/dev-explorer-backend"
```

File: .github/workflows/docker-buildx.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Build and publish docker image
+        if: ${{ !contains(steps.get_version.outputs.VERSION, '-') }}
+        uses: docker/build-push-action@v2
+        with:
+          context: docker/release
+          file: ./docker/release/Dockerfile.release
```
Comment (polarker) [inline .github/workflows/docker-buildx.yml:44]: We will need to have the folder `docker/release` as in full node

Comment (tdroxler) [reply .github/workflows/docker-buildx.yml:44]: ooooh I didn't understand this like that. let me check

Comment (polarker) [reply .github/workflows/docker-buildx.yml:44]: It basically takes the jar file and the docker folder to build a new image. So we will need to adapt the docker file a bit.

Comment (tdroxler) [reply .github/workflows/docker-buildx.yml:44]: Done, I did run a `docker build -f Dockerfile.release .` and then `docker run <image_id>` everything seems fine (I cannot to PG, but it's the same with the official docker image)

Comment (polarker) [reply .github/workflows/docker-buildx.yml:44]: I will give it a try tomorrow

File: docker/release/Dockerfile.release [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ARG RELEASE=0.0.0
+
+RUN curl -L https://github.com/alephium/explorer-backend/releases/download/v${RELEASE}/explorer-backend-${RELEASE}.jar -o /explorer-backend.jar
+
+COPY entrypoint.sh /entrypoint.sh
+
+# http
+EXPOSE 9090
+
+ENV JAVA_NET_OPTS="-Djava.net.preferIPv4Stack=true"
```
Comment (polarker) [inline docker/release/Dockerfile.release:13]: Let's add `USER nobody` before this line

**Comment by polarker** [comment]
I tested it locally, it works well. Just one minor comment.

**Review by polarker — APPROVED** [review]

**Pull request merged**